cloudinit/cmd/devel/logs.py

```diff
@@ -143,15 +144,31 @@ def _copytree_rundir_ignore_files(curdir, files):
 
 def _write_command_output_to_file(cmd, filename, msg, verbosity):
     """Helper which runs a command and writes output or error to filename."""
+    output = ""
     try:
-        out, _ = subp(cmd)
+        ensure_dir(os.path.dirname(filename))
+        output = subp(cmd)[0]
+        write_file(filename, output)
+    except ProcessExecutionError as e:
+        write_file(filename, str(e))
+        _debug("collecting %s failed.\n" % msg, 1, verbosity)
+        return str(e)
+    else:
+        _debug("collected %s\n" % msg, 1, verbosity)
+        return output
+
+
+def _stream_command_output_to_file(cmd, filename, msg, verbosity):
+    """Helper which runs a command and writes output or error to filename."""
+    try:
+        ensure_dir(os.path.dirname(filename))
+        with open(filename, "w") as f:
+            subprocess.call(cmd, stdout=f, stderr=f)
```

Comment:
`ProcessExecutionError` is defined by cloud-init, not subprocess. You need to ensure you're catching the proper exceptions.

Reply:
@a-dubs - This comment still needs to be addressed.

Reply:
Replaced `ProcessExecutionError` with `OSError` as this is the error that subp() catches to raise its `ProcessExecutionError`. This way `_stream_command_output_to_file` will only catch the same errors that `_write_command_output_to_file`, upholding the current expected functionality.